web/web.go

```diff
@@ -240,9 +241,16 @@ func MustBuildServerURL(pathPrefix string) string {
 	if err != nil {
 		panic(err)
 	}
-	hostname, err := os.Hostname()
+	hostname, err := Hostname()
 	if err != nil {
 		panic(err)
 	}
 	return fmt.Sprintf("http://%s:%s%s", hostname, port, pathPrefix)
 }
+
+func Hostname() (string, error) {
```

Comment:
Ah right, I didn't see that you already had a variable named `hostname`. If it's defined and set to `hostname()` in the same line, it would actually work, but I agree it can be confusing. Capitalizing a function means exporting it from the package though, so we want to avoid that (only export what's needed). With that in mind, I guess a name like `buildHostname` (or even `getHostname` :P) would be ok.